src/inference.jl

```diff
@@ -141,6 +144,12 @@ function __inference_check_dicttype(keyword::Symbol, ::T) where {T}
           """)
 end
 
+__inference_check_dataismissing(d) = (ismissing(d) || any(ismissing, d))
+
+# Return NamedTuple for predictions
+__inference_fill_predictions(s::Symbol, d::AbstractArray) = NamedTuple{Tuple([s])}([repeat([missing], length(d))])
+__inference_fill_predictions(s::Symbol, d::DataVariable) = NamedTuple{Tuple([s])}([missing])
```

Comment:
```suggestion
__inference_fill_predictions(s::Symbol, d::AbstractArray) = NamedTuple{(s, )}((Iterators.repeated(missing, length(d)), ))
__inference_fill_predictions(s::Symbol, d::DataVariable) = NamedTuple{(s, )}((missing, ))
```

for better perf and less memory allocs, but please double check that `Iterators.repeated` actually works